### PR TITLE
Improve gateway connection status handling in console

### PR DESCRIPTION
**Pull request by bafonins**
#### Summary

This PR addds various improvements to the gateway connection status UI.
References: https://github.com/TheThingsNetwork/lorawan-stack/issues/1086 and https://github.com/TheThingsNetwork/lorawan-stack/issues/1175
Closes https://github.com/TheThingsNetwork/lorawan-stack/issues/1326

Screenshot 2019-09-09 at 12 14 03

If no status events/stats received:
Screenshot 2019-09-24 at 14 43 33

#### Changes

- Add general event type selectors
- Update gateway stats on `gs.status.receive`, `gs.down.send`, `gs.up.receive`
- Increase gateway stats interval from 5s to **60s**
- Add uplink/downlink number formatting
- Rename `GatewayStatistics` to `GatewayConnection`. later we can split the component into `Statistics` and `Status` that can be used for https://github.com/TheThingsNetwork/lorawan-stack/issues/1155
- Add `gatewayStatus` reducer

#### Notes for Reviewers

@johanstokking According to the demo environment, some gateways receive uplinks every ~10s, imo it would be an overkill to poll for stats within the same interval. This is why I decided to increase it to 60s. Let me know what you think.

#### Notes for Reviewers

Tested with `simulate` from the cli, however it would be great to test changes with real gateway events.

### CODE REVIEW

**Comment by johanstokking** [comment]
> According to the demo environment, some gateways receive uplinks every ~10s, imo it would be an overkill to poll for stats within the same interval. This is why I decided to increase it to 60s. Let me know what you think.

I think 60 seconds is fine. Most of the realtime things (receive uplink and status) should be driven by events anyway, not (primarily) by polling stats.

File: pkg/webui/console/store/middleware/logics/gateways.js [hunk 1]
```diff
@@ -201,6 +206,10 @@ const updateGatewayStatisticsLogic = createRequestLogic({
 
     const stats = await api.gateway.stats(id)
 
+    if (!Boolean(stats.last_status_received_at)) {
+      stats.last_status_received_at = new Date()
```
Comment (kschiffer) [inline pkg/webui/console/store/middleware/logics/gateways.js]: Is this the correct assumption? Why would this prop be omitted from the response? In the proptypes it is also set as required.

Comment (bafonins) [reply pkg/webui/console/store/middleware/logics/gateways.js]: > Why would this prop be omitted from the response?

some gateways don’t send status at all. had discussion about this with @johanstokking

Comment (kschiffer) [reply pkg/webui/console/store/middleware/logics/gateways.js]: I see. Should we set `new Date()` only for heartbeat-triggered updates then?

Comment (bafonins) [reply pkg/webui/console/store/middleware/logics/gateways.js]: not sure what you mean

Comment (kschiffer) [reply pkg/webui/console/store/middleware/logics/gateways.js]: I mean that for gateways that don't send status, we would then still bump the last-seen in a 60s interval. Yet, we only safely know the gateway was seen when we received `gs.status.receive`, `gs.down.send`, `gs.up.receive`.

Comment (bafonins) [reply pkg/webui/console/store/middleware/logics/gateways.js]: @johanstokking

**Comment by kschiffer** [comment]
It's hard for me to verify in the demo environment as all gateways I have available there don't send any `last_status_received_at` prop.

Also, in these cases, if the polling interval elapses we incorrectly bump the `last seen` seen value via `stats.last_status_received_at = new Date()`, which is not desired behavior as I understand it. 

Would it be an idea to fall back to `last_uplink_received_at`?

File: pkg/webui/console/store/middleware/logics/gateways.js (hunk 1 above)
Comment (johanstokking) [reply pkg/webui/console/store/middleware/logics/gateways.js]: "Last seen" refers specifically and only to:

- `last_{status|uplink}_received_at`
- `gs.{status|up}.receive`

It should not be updated when fetching stats on the interval. If the gateway is connected but there are no status or uplinks received, it should not be implied that we've seen traffic (this diff). Still, the gateway shows connected.

Maybe "last seen" is the wrong term as it's not just that it's connected, but more "last message received"

**Comment by johanstokking** [comment]
> [...] if the polling interval elapses we incorrectly bump the `last seen` seen value via `stats.last_status_received_at = new Date()`, which is not desired behavior as I understand it.
> 
> Would it be an idea to fall back to `last_uplink_received_at`?

Sounds right. Last seen is not related to polling or the connection being there; it's purely uplink and status receive (via stats timestamps or via events).

**Pull request merged**